sendgrid_backend/mail.py

```diff
@@ -272,13 +271,54 @@ def _build_sg_mail(self, msg: EmailMessage) -> Dict:
                 )
                 # See https://github.com/sendgrid/sendgrid-nodejs/issues/843
         else:
-            personalization.subject = msg.subject
+            personalization.subject = personalizations.get("subject") or msg.subject
+
+        for header in extra_headers:
+            personalization.add_header(header)
+            
+        # write through the send_at attribute
+        if hasattr(msg, "send_at"):
+            if not isinstance(msg.send_at, int):
+                raise ValueError(
+                    "send_at must be an integer, got: {}; "
+                    "see https://sendgrid.com/docs/API_Reference/SMTP_API/scheduling_parameters.html#-Send-At".format(
+                        type(msg.send_at)
+                    )
+                )
+            personalization.send_at = msg.send_at
+
+        if hasattr(msg, "template_id"):
+            for k, v in personalizations.get("substitutions", getattr(msg, "substitutions", {})).items():
+                personalization.add_substitution(Substitution(k, v))
+            if hasattr(msg, "dynamic_template_data"):
+                if SENDGRID_5:
+                    logger.warning(
+                        "dynamic_template_data not available in sendgrid version < 6"
+                    )
+                personalization.dynamic_template_data = (
```

Comment:
Can you move this behind an `else:` block? We shouldn't be sending `dynamic_template_data` if running in sendgrid v5.  As this is currently written, we will emit the warning and also add the `dynamic_template_data`, which seems incorrect to me